### PR TITLE
Fix performance problem with file drag & drop

### PR DESCRIPTION
**Pull request by mtsgrd**
- fixes manual use of a svelte action
- pass store to DraggableFile instead of updating selection list
- refactor file selection

### CODE REVIEW

**Review by ndom91 — APPROVED** [review]

File: apps/desktop/src/lib/file/FileListItem.svelte [hunk 1]
```diff
@@ -45,22 +46,13 @@
...
-			return [file];
-		}
-	});
+	const draggable = !readonly && !isUnapplied;
 
 	let contextMenu = $state<ReturnType<typeof FileContextMenu>>();
-
 	let draggableEl: HTMLDivElement | undefined = $state();
-	let checked = $state(false);
+	let draggableFile: DraggableFile | undefined;
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/file/FileListItem.svelte]: Why does it make sense to declare this here? I don't see it used outside the effect. If it is used outside the effect, should it be a `$state()`?

Comment (mtsgrd) [reply apps/desktop/src/lib/file/FileListItem.svelte]: Thanks, that was a mistake, the variable is not needed.

File: apps/desktop/src/lib/file/FileListItem.svelte [hunk 2]
```diff
@@ -109,10 +116,11 @@
...
 	onDestroy(() => {
+		chips?.destroy();
 		if (draggableEl && animationEndHandler) {
 			draggableEl.removeEventListener('animationend', animationEndHandler);
 		}
-		draggableFiles.forEach((f) => {
+		$selectedFiles.forEach((f) => {
 			const lockedElement = document.getElementById(`file-${f.id}`);
 			if (lockedElement && animationEndHandler) {
 				lockedElement.removeEventListener('animationend', animationEndHandler);
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/file/FileListItem.svelte:126]: This doesn't guarantee that we remove all the event listeners. We ought to record all the elements that have the event added to them, and then iterate through that set and remove the event listeners added to those elements

File: apps/desktop/src/lib/vbranches/fileIdSelection.ts [hunk 3]
```diff
@@ -26,129 +29,173 @@ export function parseFileKey(fileKeyString: string): FileKey {
...
-	clear() {
-		this.value = [];
-		this.emit();
+	add(file: AnyFile, commitId?: string) {
+		this.select_many([file], commitId);
 	}
 
-	clearExcept(fileId: string, commitId?: string) {
-		this.value = [stringifyFileKey(fileId, commitId)];
+	select_many(files: AnyFile[], commitId?: string) {
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/vbranches/fileIdSelection.ts:139]: `selectMany`

File: apps/desktop/src/lib/dragging/draggable.ts [hunk 4]
```diff
@@ -175,6 +175,7 @@ function setupDragHandlers(
 		node.removeEventListener('dragstart', handleDragStart);
 		node.removeEventListener('drag', handleDrag);
 		node.removeEventListener('dragend', handleDragEnd);
+		node.removeEventListener('mousedown', handleMouseDown, { capture: false });
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/lib/dragging/draggable.ts:178]: Is this the main offending leaky event listener you were mentioning?

**Review by Caleb-T-Owens — CHANGES_REQUESTED** [review]

**Pull request merged**